pkg/kubectl/cmd/logs.go

```diff
@@ -234,26 +290,44 @@ func (o LogsOptions) Validate() error {
 }
 
 // RunLogs retrieves a pod log
-func (o LogsOptions) RunLogs() error {
-	switch t := o.Object.(type) {
-	case *api.PodList:
-		for _, p := range t.Items {
-			if err := o.getLogs(&p); err != nil {
-				return err
+func (o LogsOptions) RunLogs(out io.Writer, args []string) error {
+	if matchLen := len(o.PrefixMatchList); matchLen > 1 {
+		if matchLen > 5 {
+			o.PrefixMatchList = o.PrefixMatchList[:5]
+		}
+		first := true
+		fmt.Fprintf(out, "No pod named %s, did you mean? ", args[0])
```

Comment:
You can fetch logs from other resources besides pods, so this message is too specific.